test/adapter/json_api/key_case_test.rb

```diff
@@ -62,14 +52,14 @@ def setup
                            publish_at: @publish_at)
           @comment1.post = @post
           @comment2.post = @post
+          @error_resource = ModelWithErrors.new
+          @error_resource.errors.add(:published_at, 'must be in the future')
+          @error_resource.errors.add(:title, 'must be longer')
         end
 
         def test_success_document_key_transform_default
-          mock_request
```

Comment:
There are three main changes in this file:

* Replaced `mock_request` by `options`. ref. https://github.com/rails-api/active_model_serializers/pull/1572/files#r56795398
* Replaced `ActiveModelSerializers::Adapter::JsonApi.new(serializer)`  by `serialize_resource(@post, options(nil, serializer: PostSerializer))` so that the options are passed via the constructor and no more via the `as_json` options.
* Updated the syntax for the hash e.g. `:symbol =>` to `symbol:`, indentation, etc.